bika/lims/browser/js/bika.lims.client.js

```diff
@@ -64,3 +64,44 @@ function ClientOverlayHandler() {
       });
   }
 }
+
+function ClientSamplesView() {
+    "use strict";
+    var that = this;
+
+    that.load = function() {
+        $('a.context_action_link').click(print_samples_trigger);
+    };
+
+    function print_samples_trigger(event){
+        /*
+        This function triggers when the print button is clicked.
+        It gets all selected samples and buils the url redirections with
+        their uid so the printing function could get back the samples.
+        If no samples selected, it shows an error.
+        */
+        event.preventDefault();
+        var s_uids = [];
+        // Gets the selected samples if they are in 'scheduled sampling' state or
+        // in 'to be sampled' state
+        $("input[id^='samples_cb_']:checked").each(function(){
+            var tr = $(this).closest('tr');
+            var uid = $(tr).attr('uid');
+            var tr_state = $(tr).find('.state_title').attr('class');
+            if (tr_state.indexOf("state-scheduled_sampling") >= 0 ||
+                tr_state.indexOf("state-to_be_sampled") >= 0){
+                s_uids.push(uid);
+            }
+        });
+        // Getting the url
+        var href = $('a.context_action_link').attr('href');
+        if (s_uids.length < 1) {
+            window.location.href = $('base').attr('href') + href;
+        }
+        else {
+            href += "?items=";
+            // Add the uids in the url and redirect
+            window.location.href = $('base').attr('href') + href + s_uids.join();
+        }
+    }
+}
```

Comment:
Please remove the JS completely and handle that in the `__call__` of the `SamplesPrintView` here: https://github.com/senaite/senaite.core/blob/master/bika/lims/browser/sample/printform.py#L61

Something like this:
```python
uids = self.request.form.get("uids", [])
objs = map(api.get_object, uids)
to_print_objs = filter(lambda obj: api.get_workflow_for(obj) in ["to_be_sampled", "to_be_scheduled"], objs)
to_print_uids = map(api.get_uid, to_print_objs)
self.request.response.redirect(self.context.absolute_url() + "?items=%s" % ",".join(to_print_uids))
```